unstructured/ingest/interfaces.py

```diff
@@ -211,9 +213,25 @@ class PermissionsConfig(BaseConfig):
     tenant: t.Optional[str]
 
 
+# module-level variable to store session handle
+global_write_session_handle: t.Optional[BaseSessionHandle] = None
+
+
 @dataclass
 class WriteConfig(BaseConfig):
-    pass
+    def global_session(self):
```

Comment:
Shouldn't this live in the `BaseDestinationConnector`?

Reply:
Sessions need to be created after subprocesses are created, and within an object that is not shared between subprocesses. For example for source connectors, a global session is created in `Reader(SourceNode)` rather than `DocFactory(DocFactory)` node:
https://github.com/Unstructured-IO/unstructured/blob/30cbc420a0809d4fcde2397d9b096b1e8e638203/unstructured/ingest/pipeline/source.py#L14-L15

---

We don't have ingest docs within destination connectors, and therefore no analogue to the Reader step

So I thought creating a new `WriteConfig` object for each subprocess would be a good (and intuitive) analogue to our ingest doc based session handle approach in source connectors, that's why global session handle creation resides in WriteConfig for destination connectors.  

The pattern is that WriteConfigs have a module level (global) session handle to share within the process, in addition to WriteConfigs carrying inheritance from IngestDocSessionHandleMixin:
https://github.com/Unstructured-IO/unstructured/blob/ca0785e97860cba9191a5b68a5a8cdff4a491146/unstructured/ingest/connector/pinecone.py#L52

https://github.com/Unstructured-IO/unstructured/blob/35701dcd510f4d5f4f02b48ddcdb44eee894322d/unstructured/ingest/interfaces.py#L221-L226

Reply:
This works for now. We might want to find something else in the future as I'd like to avoid having a lot of business logic tied to the config base classes.